### PR TITLE
Raise an exception if upload report indicates failure

### PR DESCRIPTION
**Pull request by dkliban**
https://pulp.plan.io/issues/1747
closes #1747

### CODE REVIEW

**Comment by goosemania** [comment]
ok test

File: common/pulp/common/error_codes.py [hunk 1]
```diff
@@ -113,6 +113,9 @@
 PLP0043 = Error("PLP0043", _("Database 'write_concern' config can only be 'majority' or 'all'. "
                              "Refer to /etc/pulp/server.conf for proper use."), [])
 PLP0044 = Error("PLP0033", _("The target importer does not support the types from the source"), [])
+PLP0045 = Error("PLP0045", _("The importer %(importer_id)s indicated a failed response when "
```
Comment (mhrivnak) [inline common/pulp/common/error_codes.py]: oops, I just merged another PR that stole this number and 0046. ;)

File: common/pulp/common/error_codes.py [hunk 2]
```diff
@@ -113,6 +113,9 @@
...
+                             "uploading %(unit_type)s unit to repository %(repository_id)s."),
+                ['importer_id', 'unit_type', 'repository_id'])
```
Comment (mhrivnak) [inline common/pulp/common/error_codes.py]: I suggest ``repo_id`` just to match the convention everywhere else.

Comment (goosemania) [inline common/pulp/common/error_codes.py]: Fixed in several places

File: common/pulp/common/error_codes.py (hunk 1 above)
Comment (goosemania) [inline common/pulp/common/error_codes.py]: hehe, fixed)

File: server/pulp/server/managers/content/upload.py [hunk 3]
```diff
@@ -180,8 +181,12 @@ def import_uploaded_unit(repo_id, unit_type_id, unit_key, unit_metadata, upload_
...
         :param upload_id:     upload being imported
         :type  upload_id:     str
-        :return:              A SyncReport indicating the success or failure of the upload
-        :rtype:               pulp.plugins.model.SyncReport
+        :return:              A dictionary describing the success or failure of the upload. It must
+                              contain the following keys:
+                                'success_flag': bool. Indicates whether the upload was successful
+                                'summary':      json-serializable object, providing summary
+                                'details':      json-serializable object, providing details
+        :rtype:               dict
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/upload.py]: A couple of ``:raises`` entries would be helpful, both for the ``PulpCodedException`` you added, and the existing ``MissingResource`` while you're at it.

**Comment by goosemania** [comment]
ok test

**Comment by goosemania** [comment]
RHEL6 failure is unrelated to this PR

**Pull request merged**